build/android.mk

```diff
@@ -260,13 +282,30 @@ ANDROID_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES)
 $(ANDROID_LIB_BUILD): $(ANDROID_ABI_DIR)/lib%.so: $(ABI_BIN_DIR)/lib%.so | $(ANDROID_ABI_DIR)/dirstamp
 	$(Q)cp $< $@
 
+ifeq ($(ANDROID_INCLUDE_LEGACY_LIB),y)
+  ANDROID_LEGACY_LIB_DIR = $(TARGET_OUTPUT_DIR)_LEGACY/$(XCSOAR_ABI)/bin
+  ANDROID_LEGACY_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES)_legacy)
+
+  $(ANDROID_LEGACY_LIB_BUILD): $(ANDROID_LEGACY_LIB_DIR)/lib$(ANDROID_LIB_NAMES)_legacy.so
+	$(Q)cp $< $@
+
+  $(ANDROID_LEGACY_LIB_DIR)/lib$(ANDROID_LIB_NAMES)_legacy.so : ANDROID_LEGACY_FORCE
+	$(Q)$(MAKE) TARGET=$(ANDROID_ORG_TARGET) ANDROID_LEGACY=y ANDROID_INCLUDE_LEGACY_LIB=n DEBUG=$(DEBUG) USE_CCACHE=$(USE_CCACHE) V=$(V) $@
+
+  # Dummy target to force the sub-make run each time.
```

Comment:
Horrible kludge. Please not.

Reply:
Worked around it. Now the legacy lib depends on the standard lib, and forces a legacy make when the standard lib is newer, i.e. re-compiled.